object.go

```diff
@@ -9,47 +9,29 @@ package tiledb
 */
 import "C"
 import (
-	"errors"
 	"fmt"
 	"unsafe"
 
 	pointer "github.com/mattn/go-pointer"
 )
 
-// Object implements "object" managemnt in TileDB
+// ObjectType returns the object type
 // A TileDB "object" is currently either a TileDB array or a TileDB group.
-type Object struct {
-	context *Context
-	path    string
-}
-
-func NewObject(context *Context, path string) (*Object, error) {
-	if path == "" {
-		return nil, errors.New("Object path cannot be empty")
-	}
-	object := Object{
-		context: context,
-		path:    path,
-	}
-	return &object, nil
-}
-
-// Type returns the object type
-func (o *Object) Type() (ObjectType, error) {
-	var objectType C.tiledb_object_t
-	cpath := C.CString(o.path)
+func ObjectType(ctx *Context, path string) (ObjectTypeEnum, error) {
+	var objectTypeEnum C.tiledb_object_t
```

Comment:
Lets add a context != nil check to all the functions

Reply:
👍